compose/service.py

```diff
@@ -885,17 +885,19 @@ def has_host_port(binding):
 
         return any(has_host_port(binding) for binding in self.options.get('ports', []))
 
-    def pull(self, ignore_pull_failures=False):
+    def pull(self, ignore_pull_failures=False, silent=False):
```

Comment:
Maybe instead of `silent`, have a `no_progress` argument that still logs the `Pulling ...` lines, but omits the progress_stream part.